kafka-ui-react-app/src/components/Connect/Edit/Edit.tsx

```diff
@@ -109,16 +107,16 @@ const Edit: React.FC<EditProps> = ({
             <Controller
               control={control}
               name="config"
-              render={({ name, value, onChange, onBlur }) => (
-                <JSONEditor
-                  ref={register}
-                  name={name}
-                  value={value}
-                  onChange={onChange}
-                  onBlur={onBlur}
-                  readOnly={isSubmitting}
-                />
-              )}
+              render={({
+                field,
+              }: {
+                field: {
+                  name: string;
+                  onChange: () => void;
+                  value: string;
+                  onBlur: () => void;
+                };
+              }) => <JSONEditor {...field} readOnly={isSubmitting} />}
```

Comment:
You don't have to redefine `field` interface here. I believe there's `ref` property in `field` that is being passed to `JSONEditor` by spreading, it can be confusing. You can make
```tsx
render={({ field }) => (
  <JSONEditor {...field} readOnly={isSubmitting} />
)}
```